packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1705,8 +1705,20 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
                 return { ...attemptData, error };
             }
 
+            const dataStoreTree = this.disableIsolatedChannels ? summarizeResult.summary :
+                summarizeResult.summary.tree[channelsTreeName];
+            assert(dataStoreTree.type === SummaryType.Tree, "summary is not a tree");
+            const handleCount = Object.values(dataStoreTree.tree).filter(
+                (value) => value.type === SummaryType.Handle).length;
+
+            const stats: IGeneratedSummaryStats = {
```

Comment:
Please add tests for this. Preferably in this PR itself but if not, a quick follow up.